frontend/src/generate-static-html.js

```diff
@@ -1,24 +1,42 @@
 
 import React from 'react';
 import ReactDOMServer from 'react-dom/server';
+import ReactMarkdown from 'react-markdown';
 
+import Loading from './app/components/Loading';
 
-function makeStaticString(headComponent, bodyComponent) {
+function makeStaticString(
+  prepareForClient, componentName, headComponent,
+  staticBodyComponent, dynamicBodyComponent, dynamicBodyArgument
+) {
   const head = ReactDOMServer.renderToStaticMarkup(headComponent);
-  const root = ReactDOMServer.renderToStaticMarkup(bodyComponent);
+  const staticRoot = ReactDOMServer.renderToStaticMarkup(staticBodyComponent);
+  let dynamicRoot;
+  let bootstrapScript;
+  if (prepareForClient) {
+    dynamicRoot = ReactDOMServer.renderToString(dynamicBodyComponent);
+    bootstrapScript = `<script src="/static/app/bootstrap.js"></script>
+<script>${componentName.slice(0, componentName.indexOf('.'))}.default(${dynamicBodyArgument});</script>`;
+  } else if (dynamicBodyComponent) {
+    dynamicRoot = ReactDOMServer.renderToStaticMarkup(dynamicBodyComponent);
+    bootstrapScript = '';
+  }
   return `<!DOCTYPE html>
 <html>
   ${head}
   <body class="blue">
     <div class="stars"></div>
-    <div id="page-container">${root}</div>
+    <div id="page-container">${dynamicRoot}</div>
+    <div id="static-root">${staticRoot}</div>
```

Comment:
Yeah, I don't think this makes sense going forward. I think it is left over from an intermediate state where I wasn't fully rendering the server-rendered markup. I think I can remove static-root entirely now that the server-rendered markup is fully implemented.